packages/nouns-contracts/contracts/governance/NounsDAOV3Proposals.sol

```diff
@@ -395,43 +411,70 @@ library NounsDAOV3Proposals {
     }
 
     function queueOrRevertInternal(
-        NounsDAOStorageV3.StorageV3 storage ds,
+        INounsDAOExecutor timelock,
         address target,
         uint256 value,
         string memory signature,
         bytes memory data,
         uint256 eta
     ) internal {
         require(
-            !ds.timelock.queuedTransactions(keccak256(abi.encode(target, value, signature, data, eta))),
+            !timelock.queuedTransactions(keccak256(abi.encode(target, value, signature, data, eta))),
             'NounsDAO::queueOrRevertInternal: identical proposal action already queued at eta'
         );
-        ds.timelock.queueTransaction(target, value, signature, data, eta);
+        timelock.queueTransaction(target, value, signature, data, eta);
     }
 
     /**
      * @notice Executes a queued proposal if eta has passed
      * @param proposalId The id of the proposal to execute
      */
     function execute(NounsDAOStorageV3.StorageV3 storage ds, uint256 proposalId) external {
+        NounsDAOStorageV3.Proposal storage proposal = ds._proposals[proposalId];
+        INounsDAOExecutor timelock = getProposalTimelock(ds, proposal);
```

Comment:
given that we have executeOnV1, perhaps we keep this execute function more gas efficient and assume it's always the new timelock?